filewatcher/filewatcher.go

```diff
@@ -60,14 +60,27 @@ type Result struct {
 
 	ctx    context.Context
 	cancel context.CancelFunc
+
+	path      string
+	softLimit int64
+	hardLimit int64
+	logger    log.Wrapper
+	parser    Parser
 }
 
 // Get returns the latest parsed data from the file watcher.
 //
+// It might also call Parser if the file is changed from the latest parsed data.
+//
 // Although the type is interface{},
 // it's guaranteed to be whatever actual type is implemented inside Parser.
 func (r *Result) Get() interface{} {
-	return r.data.Load()
+	r.reload()
+	d, _ := r.data.Load().(*atomicData)
```

Comment:
if this type assertion cannot fail, don't make it a checked type asertion and _definitely_ don't silently ignore it

Reply:
returning nil is silently ignoring it -- it's pushing the failure into user code, where we have lost context around what the actual failure was.  What we need to know is what the type was if it wasn't *atomicData